components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/ClaimsUtil.java

```diff
@@ -206,6 +206,91 @@ public static Map<String, String> convertClaimsToOIDCDialect(OAuthTokenReqMessag
         return mappedAppClaims;
     }
 
+    /**
+     * Obtain the SP requested claims from the claims sent in the token.
+     *
+     * @param requestMsgCtx Token request message context.
+     * @param userClaims    Claims in OIDC dialect.
+     * @return              SP requested claims which are available in the userClaims.
+     * @throws IdentityApplicationManagementException   Identity Application Management Exception.
+     * @throws IdentityException                        Identity Exception.
+     */
+    private static Map<String, String> getSPRequestedClaimsInOIDCDialect(OAuthTokenReqMessageContext requestMsgCtx,
+                                                                 Map<String, String> userClaims) throws
+            IdentityApplicationManagementException, IdentityException {
+
+        Map<String, String> mappedAppClaims = new HashMap<>();
+
+        if (log.isDebugEnabled()) {
+            StringBuilder claims = new StringBuilder();
+            for (String key : userClaims.keySet()) {
+                claims.append(key).append(",");
+            }
+            log.debug("Obtaining requested claims for user: " + requestMsgCtx
+                    .getAuthorizedUser() + ", client id:" + requestMsgCtx.getOauth2AccessTokenReqDTO().getClientId()
+                    + ", converting claims: " + claims.toString());
+        }
+
+        ServiceProvider serviceProvider = getServiceProvider(requestMsgCtx);
+
+        if (serviceProvider == null) {
+            return mappedAppClaims;
+        }
+        ClaimMapping[] spClaimMappings = serviceProvider.getClaimConfig().getClaimMappings();
+        if (spClaimMappings == null || !(spClaimMappings.length > 0)) {
```

Comment:
Shall we use StringUtils.isBlank?

Reply:
Cannot use StringUtils.isBlank() as ServiceProvider object is not a String. Only Strings can be used with StringUtils.isBlank().